### ISSUE TITLE
International Country Phone and Address Issues

### ISSUE THREAD

**Issue by radams22** [SparkDevNetwork/Rock#5468]
## Please go through all the tasks below

* [x] Check this box only after you have successfully completed both the above tasks

## Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.
Tell us what you see!https://github.com/SparkDevNetwork/Rock/assets/72230330/364ace88-4398-4462-bd38-5df8439003de

## Expected Behavior
Expected Behavior would be that the country code would be updated based on what is entered into the workflow and it not default back to the first selection. For the address issue I mentioned, I would think it should allow you to default it to any or no country so the person filling out the form can select the right country and it show the correct fields for that address in that country. And AZ would not remain in the state/province field when switching between the United States and another country selection.

## Actual Behavior
Actual Behavior - It doesn't update the country code on the phone number based on what is selected on the form.It always reverts back to the United States and leaves AZ in the State/Province field. 

## Steps to Reproduce
I explained the workflow in the video. But you will need to add a country code to the defined values so you can see more of a selection on the workflow phone number selection, as well as change the international address attribute to yes. Then create a workflow to collect the person, phone number, and address and submit it after changing the country code to the second option in the list. 

## Rock Version
13.7, 14, 15

## Client Culture Setting
en-US

**Comment by radams22** [comment]
There is another github issue that I didn't see originally when I posted this one, but it's the exact same issue from 4 years ago.
https://github.com/SparkDevNetwork/Rock/issues/3760#issue-459319849

**Comment by nairdo** [comment]
@radams22  You're right that the phone number issue is the same as 3760, but we'll leave this open as we work on the "AZ" state/province field issue.

**Comment by MrUpsideDown** [comment]
@radams22 As noted, the state/province field issue has been fixed in v16.1. The phone country code issue is more complex, so that fix has been deferred to v17 and will be tracked by issue #3670.
